src/wasm-lib/kcl-test-server/src/lib.rs

```diff
@@ -178,7 +178,7 @@ async fn snapshot_endpoint(body: Bytes, state: ExecutorContext) -> Response<Body
     // Let users know if the test is taking a long time.
     let (done_tx, done_rx) = oneshot::channel::<()>();
     let timer = time_until(done_rx);
-    let snapshot = match state.execute_and_prepare_snapshot(&program, id_generator).await {
+    let snapshot = match state.execute_and_prepare_snapshot(&program, id_generator, None).await {
```

Comment:
If you avoid modifying the `execute_and_prepare_snapshot` method, this PR diff will be much smaller (just make a new method that lets you take the optional param instead, and make the existing one set it to None)